### PR TITLE
add sync ecommerce on course verification

### PR DESCRIPTION
**Pull request by inlanger**

### CODE REVIEW

File: ecommerce/extensions/api/serializers.py [hunk 1]
```diff
@@ -336,6 +336,12 @@ def _flatten(self, attrs):
         return {attr['name']: attr['value'] for attr in attrs}
 
 
+class EdevateAtomicPublicationSerializer(AtomicPublicationSerializer):  # pylint: disable=abstract-method
+    def __init__(self, *args, **kwargs):
+        super(EdevateAtomicPublicationSerializer, self).__init__(*args, **kwargs)
+        self.partner = kwargs['context'].pop('partner', None)
```
Comment (CrowbarKZ) [inline ecommerce/extensions/api/serializers.py:342]: i am a bit out of the loop. Why do we pop `partner` here?

File: ecommerce/extensions/api/v2/views/publication.py [hunk 2]
```diff
@@ -42,3 +44,22 @@ def _save_and_publish(self, data, course_id=None):
                 content = serializer.data
                 content['message'] = message if message else None
                 return Response(content, status=status.HTTP_201_CREATED if created else status.HTTP_200_OK)
+
+
+class EdevateAtomicPublicationView(generics.CreateAPIView, generics.UpdateAPIView):
+    """Attempt to save and publish a Course and associated products.
+
+    If either fails, the entire operation is rolled back. This keeps Otto and the LMS in sync.
```
Comment (CrowbarKZ) [inline ecommerce/extensions/api/v2/views/publication.py]: probably better to change docstring to mention what is different from the normal AtomicPublicationView

File: ecommerce/extensions/api/v2/views/publication.py [hunk 3]
```diff
@@ -42,3 +44,22 @@ def _save_and_publish(self, data, course_id=None):
...
+    """
+    permission_classes = (AllowAny,)
```
Comment (CrowbarKZ) [inline ecommerce/extensions/api/v2/views/publication.py:54]: Isn't it a problem that we don't require any permissions?

File: ecommerce/extensions/api/serializers.py (hunk 1 above)
Comment (inlanger) [reply ecommerce/extensions/api/serializers.py:342]: if request data already has partner we get it here

File: ecommerce/extensions/api/v2/views/publication.py (hunk 3 above)
Comment (inlanger) [reply ecommerce/extensions/api/v2/views/publication.py:54]: endpoint is not public so we don't need any permissions

**Review by CrowbarKZ — APPROVED** [review]

**Pull request merged**